sky/check.py

```diff
@@ -10,10 +10,20 @@
 
 
 # TODO(zhwu): add check for a single cloud to improve performance
-def check(quiet: bool = False, verbose: bool = False) -> None:
+def check(
+    quiet: bool = False,
+    verbose: bool = False,
+    user_specified_clouds: Optional[Tuple[str]] = None,
+) -> None:
     echo = (lambda *_args, **_kwargs: None) if quiet else click.echo
     echo('Checking credentials to enable clouds for SkyPilot.')
 
+    user_specified_clouds_set = set()
+    if user_specified_clouds is not None:
+        for cloud in user_specified_clouds:
+            clouds.CLOUD_REGISTRY.from_str(cloud)  # verify cloud
+            user_specified_clouds_set.add(cloud.lower())
+
     enabled_clouds = []
```

Comment:
We should cache previously enabled clouds. Else only the new cloud checked is enabled. i.e., if a user previously had AWS, Az and GCP enabled and they run `sky check aws`, currently in this PR Az and GCP will get disabled. They should stay enabled.

e.g.,
```
# Before, all clouds enabled
(base) ➜  ~ sky launch
I 02-27 18:31:46 optimizer.py:691] == Optimizer ==
I 02-27 18:31:46 optimizer.py:702] Target: minimizing cost
I 02-27 18:31:46 optimizer.py:714] Estimated cost: $0.4 / hour
I 02-27 18:31:46 optimizer.py:714]
I 02-27 18:31:46 optimizer.py:837] Considered resources (1 node):
I 02-27 18:31:46 optimizer.py:907] -------------------------------------------------------------------------------------------------
I 02-27 18:31:46 optimizer.py:907]  CLOUD    INSTANCE          vCPUs   Mem(GB)   ACCELERATORS   REGION/ZONE     COST ($)   CHOSEN
I 02-27 18:31:46 optimizer.py:907] -------------------------------------------------------------------------------------------------
I 02-27 18:31:46 optimizer.py:907]  AWS      m6i.2xlarge       8       32        -              us-east-1       0.38          ✔
I 02-27 18:31:46 optimizer.py:907]  Azure    Standard_D8s_v5   8       32        -              eastus          0.38
I 02-27 18:31:46 optimizer.py:907]  IBM      bx2-8x32          8       32        -              us-east         0.38
I 02-27 18:31:46 optimizer.py:907]  GCP      n2-standard-8     8       32        -              us-central1-a   0.39
I 02-27 18:31:46 optimizer.py:907]  Lambda   gpu_1x_a10        30      200       A10:1          us-east-1       0.60
I 02-27 18:31:46 optimizer.py:907] -------------------------------------------------------------------------------------------------
I 02-27 18:31:46 optimizer.py:907]
Launching a new cluster 'sky-7ddc-romilb'. Proceed? [Y/n]:

# Run sky check on single cloud

(base) ➜  ~ sky check aws
Checking credentials to enable clouds for SkyPilot.
  AWS: enabled

To enable a cloud, follow the hints above and rerun: sky check
If any problems remain, refer to detailed docs at: https://skypilot.readthedocs.io/en/latest/getting-started/installation.html

🎉 Enabled clouds 🎉
  ✔ AWS

# Other clouds now disabled. They should remain enabled.
(base) ➜  ~ sky launch
I 02-27 18:33:46 optimizer.py:691] == Optimizer ==
I 02-27 18:33:46 optimizer.py:702] Target: minimizing cost
I 02-27 18:33:46 optimizer.py:714] Estimated cost: $0.4 / hour
I 02-27 18:33:46 optimizer.py:714]
I 02-27 18:33:46 optimizer.py:837] Considered resources (1 node):
I 02-27 18:33:46 optimizer.py:907] ------------------------------------------------------------------------------------------
I 02-27 18:33:46 optimizer.py:907]  CLOUD   INSTANCE      vCPUs   Mem(GB)   ACCELERATORS   REGION/ZONE   COST ($)   CHOSEN
I 02-27 18:33:46 optimizer.py:907] ------------------------------------------------------------------------------------------
I 02-27 18:33:46 optimizer.py:907]  AWS     m6i.2xlarge   8       32        -              us-east-1     0.38          ✔
I 02-27 18:33:46 optimizer.py:907] ------------------------------------------------------------------------------------------
I 02-27 18:33:46 optimizer.py:907]
Launching a new cluster 'sky-2bfa-romilb'. Proceed? [Y/n]:
```

Reply:
Nice catch! Also now tracking disabled clouds to make sure we can prune clouds that can no longer be used. Otherwise, solely caching the previously enabled clouds would lead to an ever increasing list of enabled clouds